### DISCUSSION TITLE
Breaking changes and the road to GA

### DISCUSSION THREAD

**Discussion by madskristensen** [VsixCommunity/Community.VisualStudio.Toolkit#81] [General]
Hi everybody,

As the Toolkit project progresses and more discussions are happening, some breaking changes will have to be introduced. Here's what's going on:

1. New goal to make the Toolkit a viable replacement for the DTE for new extensions
    - Not for existing extensions using the DTE already
2. More concepts have been added to match the DTE
    - Help needed to identify gaps between the toolkit and the DTE
3. The API surface has moved changed a bit due to renaming and refactoring
4. `-pre` has been appended to all NuGet package versions since more breaking changes are likely before GA (not just 17.0)
    - Older versions of the toolkit will be unlisted on NuGet 
5. Aiming for GA at the same time VS 2022 goes to GA
    - Whenever that is
    - The `-pre` moniker will be removed from the packages at this time

Please help test the latest `-pre` packages recently uploaded to NuGet. Your help is much appreciated.

Thoughts, concerns, questions?

**Comment by StevenRasmussen** [comment]
I think that all makes sense.  I've seen you making quite a few changes lately which is great!  One thing that perhaps could use some clarification is with regards to issues that are raised.  I see you (or others) often create an issue first and then I see a PR submitted shortly thereafter.  Perhaps it would be prudent to start adding labels to issues so that we know what the status is?  Or at least adding a comment that this is already being worked on and a PR is coming shortly?  The only reason I say this is that I'm happy to help out where I can but it's unclear to me if the issue being added is just a formality to a PR that's coming, or if it's something that you would like/accept a PR for.

**Reply by madskristensen** [reply]
Great points about the issues. I open them and then sometimes get to work on them sooner than I think. I like the label idea
